src/overlays/actors/ovl_En_Vali/z_en_vali.c

```diff
@@ -76,71 +82,732 @@ static ColliderCylinderInit D_80B288F0 = {
+            // Cannot be %
+            effectPos.x = this->actor.world.pos.x + (((temp_v0 & 2) != 0) ? 12.0f : -12.0f);
+            effectPos.z = this->actor.world.pos.z + (((temp_v0 & 1) != 0) ? 12.0f : -12.0f);
+
+            EffectSsEnIce_SpawnFlyingVec3f(globalCtx, &this->actor, &effectPos, 150, 150, 150, 250, 235, 245, 255,
+                                           (Rand_ZeroOne() * 0.2f) + 1.3f);
+        }
+    } else if (this->timer == 0) {
+        this->actor.velocity.y += 1.0f;
+        if (Math_StepToF(&this->actor.world.pos.y, this->actor.floorHeight, this->actor.velocity.y)) {
+            EnVali_SetupDivideAndDie(this, globalCtx);
+            this->actor.colorFilterTimer = 0;
+        }
+    }
+}
+
+void EnVali_ReturnToLurk(EnVali* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+
+    if (Math_SmoothStepToF(&this->actor.world.pos.y, this->actor.home.pos.y, 0.5f, 15.0f, 0.1f) < 0.01f) {
+        EnVali_SetupLurk(this);
+    }
+}
+
+void EnVali_UpdateDamage(EnVali* this, GlobalContext* globalCtx) {
+    if (this->bodyCollider.base.acFlags & AC_HIT) {
+        this->bodyCollider.base.acFlags &= ~AC_HIT;
+        func_80035650(&this->actor, &this->bodyCollider.info, 1);
+
+        if ((this->actor.colChkInfo.damageEffect != BARI_DMGEFF_NONE) || (this->actor.colChkInfo.damage != 0)) {
+            if (Actor_ApplyDamage(&this->actor) == 0) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_BARI_DEAD);
+                func_80032C7C(globalCtx, &this->actor);
+                this->actor.flags &= ~1;
+            } else if ((this->actor.colChkInfo.damageEffect != BARI_DMGEFF_STUN) &&
+                       (this->actor.colChkInfo.damageEffect != BARI_DMGEFF_SLINGSHOT)) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_BARI_DAMAGE);
+            }
+
+            if (this->actor.colChkInfo.damageEffect == BARI_DMGEFF_STUN) {
+                if (this->actionFunc != EnVali_Stunned) {
+                    EnVali_SetupStunned(this);
+                }
+            } else if (this->actor.colChkInfo.damageEffect == BARI_DMGEFF_SWORD) {
+                if (this->actionFunc != EnVali_Stunned) {
+                    Actor_SetColorFilter(&this->actor, 0x4000, 150, 0x2000, 30);
+                    this->actor.params = BARI_TYPE_SWORD_DAMAGE;
+                    EnVali_SetupAttacked(this);
+                } else {
+                    EnVali_SetupRetaliate(this);
+                }
+            } else if (this->actor.colChkInfo.damageEffect == BARI_DMGEFF_FIRE) {
+                EnVali_SetupBurnt(this);
+            } else if (this->actor.colChkInfo.damageEffect == BARI_DMGEFF_ICE) {
+                EnVali_SetupFrozen(this);
+            } else if (this->actor.colChkInfo.damageEffect == BARI_DMGEFF_SLINGSHOT) {
+                if (this->slingshotReactionTimer == 0) {
+                    this->slingshotReactionTimer = 20;
+                }
+            } else { // Only DMGEFF_NONE
```

Comment:
I think this comment could be misleading since technically `damageEffect` could be another value and end up here in modding scenarios.

Reply:
Okay, although I would have thought that any modifications to `damageEffect` would be changing these nested if-elses anyway!